src/dags/bouwstroompunten.py

```diff
@@ -89,89 +91,98 @@ def get_data():
-        bash_command="ogr2ogr -f 'PGDump' "
-        "-t_srs EPSG:28992 "
-        f"-nln {dag_id} "
-        f"{tmp_dir}/{dag_id}.sql {data_file} "
-        "-lco GEOMETRY_NAME=geometry ",
-    )
-
-    # 5. Create TABLE
-    create_table = BashOperator(
-        task_id="create_table",
-        bash_command=f"psql {pg_params()} < {tmp_dir}/{dag_id}.sql",
+    import_data = Ogr2OgrOperator(
+        task_id="import_data",
+        target_table_name=f"{dag_id}_{dag_id}_new",
+        input_file=data_file,
+        s_srs=None,
+        fid="ogc_fid",
+        auto_detect_type="YES",
+        mode="PostgreSQL",
+        db_conn=db_conn,
     )
 
-    # 6. Drop Exisiting TABLE
+    # 5. Drop Exisiting TABLE
     drop_tables = PostgresOperator(
         task_id="drop_existing_table",
         sql=[
             f"DROP TABLE IF EXISTS {dag_id}_{dag_id} CASCADE",
         ],
     )
 
-    # 7. Rename COLUMNS based on Provenance
+    # 6. Rename COLUMNS based on provenance (if specified)
     provenance_translation = ProvenanceRenameOperator(
-        task_id="rename_columns", dataset_name=dag_id, pg_schema="public"
+        task_id="rename_columns",
+        dataset_name=dag_id,
+        prefix_table_name=f"{dag_id}_",
+        postfix_table_name="_new",
+        rename_indexes=False,
+        pg_schema="public",
     )
 
-    # 8. Rename TABLE
+    # 7. Rename TABLE
     rename_table = PostgresTableRenameOperator(
         task_id="rename_table",
-        old_table_name=dag_id,
+        old_table_name=f"{dag_id}_{dag_id}_new",
         new_table_name=f"{dag_id}_{dag_id}",
     )
 
-    # 9. RE-define PK(see step 4 why)
+    # 8. RE-define PK(see step 4 why)
     redefine_pk = PostgresOperator(
         task_id="re-define_pk",
         sql=ADD_PK,
-        params=dict(tablename=f"{dag_id}_{dag_id}"),
+        params={"tablename": f"{dag_id}_{dag_id}"},
     )
 
     # PREPARE CHECKS
     count_checks.append(
         COUNT_CHECK.make_check(
             check_id="count_check",
             pass_value=25,
-            params=dict(table_name=f"{dag_id}_{dag_id}"),
+            params={"table_name": f"{dag_id}_{dag_id}"},
             result_checker=operator.ge,
         )
     )
 
     geo_checks.append(
         GEO_CHECK.make_check(
             check_id="geo_check",
-            params=dict(
-                table_name=f"{dag_id}_{dag_id}",
-                geotype=["POINT"],
-            ),
+            params={
+                "table_name": f"{dag_id}_{dag_id}",
+                "geotype": ["POINT"],
+            },
             pass_value=1,
         )
     )
 
     total_checks = count_checks + geo_checks
 
-    # 10. RUN bundled CHECKS
+    # 9. RUN bundled CHECKS
     multi_checks = PostgresMultiCheckOperator(task_id="multi_check", checks=total_checks)
 
-    # 11. Grant database permissions
+    # 10. Grant database permissions
     grant_db_permissions = PostgresPermissionsOperator(task_id="grants", dag_name=dag_id)
 
 
 (
```

Comment:
This should be indented one level more. The workflow specification (how the tasks are linked together) should be part of the `with` statement.

Reply:
:+1: